app/teams/teams.go

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package teams
+
+import (
+	"context"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/shared/i18n"
+)
+
+func (ts *TeamService) CreateTeam(team *model.Team) (*model.Team, error) {
+	team.InviteId = ""
+	rteam, err := ts.store.Save(team)
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err := ts.createDefaultChannels(rteam.Id); err != nil {
+		return nil, err
+	}
+
+	return rteam, nil
+}
+
+func (ts *TeamService) GetTeam(teamID string) (*model.Team, error) {
+	team, err := ts.store.Get(teamID)
+	if err != nil {
+		return nil, err
+	}
+
+	return team, nil
+}
+
+// CreateDefaultChannels creates channels in the given team for each channel returned by (*App).DefaultChannelNames.
+//
+func (ts *TeamService) createDefaultChannels(teamID string) ([]*model.Channel, error) {
+	displayNames := map[string]string{
+		"town-square": i18n.T("api.channel.create_default_channels.town_square"),
+		"off-topic":   i18n.T("api.channel.create_default_channels.off_topic"),
+	}
+	channels := []*model.Channel{}
+	defaultChannelNames := ts.DefaultChannelNames()
+	for _, name := range defaultChannelNames {
+		displayName := i18n.TDefault(displayNames[name], name)
+		channel := &model.Channel{DisplayName: displayName, Name: name, Type: model.ChannelTypeOpen, TeamId: teamID}
+		// We should use the channel service here (coming soon). Ideally, we should just emit an event
```

Comment:
I was of the understanding that we would keep all "Channel" related stuff inside app and move general stuff which are not tied to channels outside. Would there be a need to move channels outside as well?

Reply:
In the end picture we may have a channels package which will be responsible of channel related stuff. The idea here is to make a connection between team and channels. Let's say we want to create a team from Boards path, we should just ask teams package to create the team then it should notify/call required entities and complete the process.